server/sonar-server/src/main/java/org/sonar/server/component/index/ComponentIndexer.java

```diff
@@ -91,11 +95,18 @@ public void deleteByProjectUuid(String uuid) {
   }
 
   private void deleteComponentsByProjectUuid(String projectUuid) {
-    esClient
-      .prepareDelete(INDEX_COMPONENTS, TYPE_COMPONENT, projectUuid)
-      .setRouting(projectUuid)
-      .setRefresh(true)
-      .get();
+    BulkIndexer.delete(esClient, INDEX_COMPONENTS, projectUuidSearch(projectUuid));
+  }
+
+  private SearchRequestBuilder projectUuidSearch(String projectUuid) {
```

Comment:
Having three private methods breaks readability IMO. `projectUuidFilter()` and `projectUuidSearch()` are used only once and so do not aim to share code. The following seems more simple for me because I can read the full Elasticsearch request : 
```
private void deleteComponentsByProjectUuid(String projectUuid) {
    BulkIndexer.delete(esClient, INDEX_COMPONENTS, esClient.prepareSearch(INDEX_COMPONENTS)
      .setQuery(boolQuery()
        .filter(
          termQuery(ComponentIndexDefinition.FIELD_PROJECT_UUID, projectUuid))));
  }
```